src/TodoContext/TodoContext.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { filterForTodos } from '../services/chooseFilterForTodos';
+import { getTodos } from '../api/todos';
+import { USER_ID } from '../variables/UserID';
+import { Todo } from '../types/Todo';
+import { TodoContext } from '../types/TodoContext.1';
+import { wait } from '../utils/fetchClient';
+
+export const TodosContext = React.createContext<TodoContext>({
+  todos: [],
+  addTodo: () => { },
+  setCompleted: () => { },
+  makeAllCompleted: () => { },
+  query: '',
+  setQuery: () => { },
+  filteredTodos: [],
+  deleteCompletedTodos: () => { },
+  deleteTodo: () => { },
+  saveEditingTitle: () => { },
+  errorMessage: '',
+  setErrorMessage: () => { },
+});
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Error');
+
+        wait(3000).then(() => setErrorMessage(''));
```

Comment:
agree with Serhiy

put setTimeout inside `Error` component, you can use `useEffect` hook for this as well

in general you will have different kind of errors, and this implementation will work the same way with all errors

currently you have this 3000 delay only when load Todos